metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_common.py

```diff
@@ -707,6 +850,182 @@ def filter_nodes(self, all_nodes: List[DBTNode]) -> List[DBTNode]:
+                                if upstream_node_name in all_nodes_map
+                            ]
+                            if upstream_node.is_ephemeral_model()
+                            for cte_name in _get_dbt_cte_names(
+                                upstream_node.name, schema_resolver.platform
+                            )
+                        },
+                    )
+                except Exception as e:
+                    sql_result = SqlParsingResult.make_from_error(e)
+                else:
+                    sql_result = sqlglot_lineage(
+                        preprocessed_sql, schema_resolver=schema_resolver
+                    )
+
+            # Save the column lineage.
+            if self.config.include_column_lineage and sql_result:
+                # We only save the debug info here. We're report errors based on it later, after
+                # applying the configured node filters.
+                node.cll_debug_info = sql_result.debug_info
+
+                if sql_result.column_lineage:
+                    node.upstream_cll = [
+                        DBTColumnLineageInfo(
+                            upstream_dbt_name=target_platform_urn_to_dbt_name[
+                                upstream_column.table
+                            ],
+                            upstream_col=upstream_column.column,
+                            downstream_col=column_lineage_info.downstream.column,
+                        )
+                        for column_lineage_info in sql_result.column_lineage
+                        for upstream_column in column_lineage_info.upstreams
+                        # Only include the CLL if the table in in the upstream list.
+                        if target_platform_urn_to_dbt_name.get(upstream_column.table)
+                        and target_platform_urn_to_dbt_name[upstream_column.table]
+                        in node.upstream_nodes
+                    ]
+
+            # If we didn't fetch the schema from the graph, use the inferred schema.
+            if not schema_fields and sql_result and sql_result.column_lineage:
+                schema_fields = [
+                    SchemaField(
+                        fieldPath=column_lineage.downstream.column,
+                        type=column_lineage.downstream.column_type
+                        or SchemaFieldDataType(type=NullTypeClass()),
+                        nativeDataType=column_lineage.downstream.native_column_type
+                        or "",
+                    )
+                    for column_lineage in sql_result.column_lineage
+                ]
+
+            # Conditionally add the inferred schema to the schema resolver.
+            if not added_to_schema_resolver and target_node_urn and schema_fields:
+                schema_resolver.add_raw_schema_info(
+                    target_node_urn, self._to_schema_info(schema_fields)
+                )
```

Comment:
Feels like this could be embedded in the above `if` statement instead -- that's where `schema_fields` is set for this right? Or do you expect we might set `schema_fields` other places in the future, after `added_to_schema_resolver` is set

Reply:
we use the inferred schema_fields in two places, so can't combine it into the above statement

I stopped reusing the schema_fields to make the code a bit more clear